bot/cogs/watchchannels/watchchannel.py

```diff
@@ -330,10 +336,3 @@ def cog_unload(self) -> None:
         self.log.trace("Unloading the cog")
         if self._consume_task and not self._consume_task.done():
             self._consume_task.cancel()
-            try:
-                self._consume_task.result()
```

Comment:
I don't agree with this. You've completely removed the fetching of the task result. The done callback approach I suggested in the issue is better because it allows for exceptions to be retrieved in addition to checking if the task was cancelled.